src/handle_finalize.c

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ *
+ * ██╗  ██╗██╗██╗     ███╗   ██╗
+ * ██║ ██╔╝██║██║     ████╗  ██║
+ * █████╔╝ ██║██║     ██╔██╗ ██║
+ * ██╔═██╗ ██║██║     ██║╚██╗██║
+ * ██║  ██╗██║███████╗██║ ╚████║
+ * ╚═╝  ╚═╝╚═╝╚══════╝╚═╝  ╚═══╝
+ *
+ * Kiln Ethereum Ledger App
+ * (c) 2022-2024 Kiln
+ *
+ * contact@kiln.fi
+ ********************************************************************************/
+
+#include "kiln_plugin.h"
+
+void handle_finalize(ethPluginFinalize_t *msg) {
+    context_t *context = (context_t *) msg->pluginContext;
+
+    msg->uiType = ETH_UI_TYPE_GENERIC;
+
+    switch (context->selectorIndex) {
+        case KILN_V1_DEPOSIT:
```

Comment:
nit: I'd favor semantic grouping over technical grouping here, and keep v1 and v2 separate, even if the number of screens is the same. If 1 day we add an extra screen specific to v2 for example, we have more work to do.